example/javascriptlib/basic/1-simple/build.mill

```diff
@@ -1,27 +1,36 @@
 package build
+
 import mill._, javascriptlib._
 
-object foo extends TypeScriptModule {
-  object bar extends TypeScriptModule {
-    def npmDeps = Seq("immutable@4.3.7")
-  }
-}
+object foo extends JestModule {
+  def npmDeps = Seq("immutable@4.3.7")
 
-object qux extends TypeScriptModule {
-  def moduleDeps = Seq(foo, foo.bar)
+  override def mainFilePath =
+    Task { compile()._1.path / millSourcePath.last / "src" / mainFileName() }
```

Comment:
Why do we need to override this?

Reply:
for projects with nested directories, tsc will recreate the directories and compile attached js output within.
so a project with src/, test/ will have src/, test/ in out/**/compile.dest/

the original setup does not account for this.

Reply:
millSourcePath.last looks like the wrong eay to implement what you described, as it only returns the last segment. What if it is nested two levels deep?

Should we push the change upstream into TypescriptModule instead?

Reply:
my mistake tsconfig will dump code into: /out/foo/compile.dest/javascript/foo/...
hence the `millSourcePath.last `

Also since the tsconfig compiler options is fully customizable, ts configurations will vary and compile output dir will also vary.

I can update the tsconfig from foo, so the main file is at a different directory (/compile.dest/javascript/....) (without the foo after javascript). But I don't see the need, projects that use will most likely use varying tsconfigs and will have varying mainFilePaths

Reply:
no longer depends on millSourcePath.last